### ISSUE TITLE
Request for getting details of a new separate test bed for traffic testing(Throughput) & long run testing.

### ISSUE THREAD

**Issue by shanjanaWL** [magma/magma#9920]
Please provide details of a new separate test bed for traffic testing(Throughput) & long run testing.

**Comment by arun-magma** [comment]
Setup readiness final leg of work is in progress. May i know what are the tests planned to run & for how long duration ?

**Comment by NeelimaKalpana** [comment]
Hi Arun,
I will look into this and update the requirements.

**Comment by NeelimaKalpana** [comment]
Hi Arun,

Kindly provide one single TC where we can cover all 5g procedures as per mvp.
Test Case should cover :
Connection=>Registration=>PDU-Establishment=>Traffic=>UE Idle-Mode=>Paging=>Traffic(After UE-Active with Paging)=>UE Idle-Mode=>Service-Request=>Traffic(After UE-Active with service-Request )=>PDU-Release=>Deregistration.

We should execute this TC for about 24hr and more for long run.
Kindly provide access to the complete setup for Wavelabs Test Team.

**Comment by arun-magma** [comment]
This looks to be a Functional E2E scenario coverage, how do you want to run it for 24hrs ? 
Repeat the same flows in loop ?
Scale and run ? ( May not need long run to verify scale) 
Run traffic part alone for long ?
or any other periodic Mix actions to be planned ?

Shall we manage to run this in existing SA emulator ? as bringing up another one may take time & resources.

**Comment by NeelimaKalpana** [comment]
Hi Arun,

1.Repeat the same flows in loop ?
Neelima:Yes, But we can design TC by Mixing multiple procedures like:
 TC1: Basic call flow , and after PDU Success send traffic for some ling duration we can have idle-mode case switch to Active-Mode with signalling and then send traffic for some long duration.
TC2: With basic flow we can add paging.
2.Scale and run ? ( May not need long run to verify scale)
Neelima:We Need all MVP requirements in a single testcases and check for multiple UE's. This can be taken as scale test and I can run it multiple times for long run.
Run traffic part alone for long ?
Neelima: We can have a TC for this also
or any other periodic Mix actions to be planned ?
Neelima: As reply to 1st quire, its good to have multiple TC with Mix of multiple MVP requirement.

**Comment by arun-magma** [comment]
As discussed in call here are more insights about TeraVm Platform and this ZH Ticket.

1. TeraVM currently doesn't support Simultaneous executions on single Ng40 (TeraVm Emulator) Machine, It's not manageable atm as execution seize/reserves all key function emulations, Networking resources and Pkt capture.
2. User Access Control is coming up soon (Awaiting Viavi ETA ?) on this Control center UIs, which would let open accounts and manage UI feature authorization, this would still not allow parallel executions. In a sense a default account (single user) is being used as parallel executions is not in place.

The ZH Ticket currently have two diff Tasks in it.

1. Prepare a new scripts with all default mentioned flows in it, let's work on to finalize the flows, once ready we can start run for few hours in current emulator (.88 machine) itself & validate the stability of the AGW code, this Might Unblock Dec-30th Targets.
2. Parallel environment for Long runs (24Hrs & beyond) which needs a new TeraVM emulator, as I explained bringing up a new emulator needs time & resources (Licenses, Compute, Networking & Sys Integ activities), Currently active TeraVM Emulators are occupied & aligned to different requirements, we will evaluate the feasibility for new one and come back.

CC: @Gosh & @vaishaliCI

**Comment by arun-magma** [comment]
Hi @NeelimaKalpana: We are discussing too many Mid session scenarios in this ticket, and Moreover as per title these are moving out of scope of this ticket, Lets have separate tickets for Mid session performance cases with detail Test plan ( as per Template), this would really help focus on scripting it better.
Add details include which Mid session scenario or flow need to be continued/repeated for longer duration, this would help to script & keep long duration control parameters as configurable.

[Below two cases are valid real UE scenario, will try start work on it, Just add info requested above in a separate ticket, Thanks!]
TC1: Basic call flow , and after PDU Success send traffic for some ling duration we can have idle-mode case switch to Active-Mode with signalling and then send traffic for some long duration.
TC2: With basic flow we can add paging.

**Comment by NeelimaKalpana** [comment]
Thank you Arun for the update.
For Mid session scenarios I will raise a ticket.

In current TVM-GUI
TC7 , TC7a ,TC7b ,TC7c , TC9 ,TC9a ,TC 9b ,TC8 ,TC8a, TC50a,TC50b,TC50c.
We need all above TC to run in sequence to run in long run of 12 hrs and above.

Due to time constraint we want to get started long run with the above TC's once all TC for current mile stone are complete we can modify above TC or make a new TC with long traffic and different traffic packets and mix and match of multiple-UE , Multi-session and multi-procedures involved.

Please do the needful .

**Comment by arun-magma** [comment]
Will review how to connect all these scenarios and make a single script, will comeback soon on this.

**Comment by rupa455wavelabs** [comment]
Hi @arun-magma, Can you please update on this.
